public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -131,6 +131,23 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
     return of({ state: LoadingState.Done, data: [] });
   }
 
+  targetContainsTemplate(query: AzureMonitorQuery) {
+    if (query.subscription && this.templateSrv.variableExists(query.subscription)) {
+      return true;
+    }
+
+    let subQuery;
+    if (query.queryType === AzureQueryType.AzureMonitor) {
+      subQuery = JSON.stringify(query.azureMonitor);
+    } else if (query.queryType === AzureQueryType.LogAnalytics) {
+      subQuery = JSON.stringify(query.azureLogAnalytics);
+    } else if (query.queryType === AzureQueryType.AzureResourceGraph) {
+      subQuery = JSON.stringify([query.azureResourceGraph, query.subscriptions]);
+    }
+
+    return !!(subQuery && this.templateSrv.variableExists(subQuery));
```

Comment:
nit: you don't need to convert to a boolean with `!!`

Reply:
I believe we do, but I guess it should be enough to convert just the subQuery. Will fix